src/main/java/io/ballerina/plugins/idea/sdk/BallerinaSdkUtils.java

```diff
@@ -0,0 +1,408 @@
+                if (files != null) {
+                    for (File file : files) {
+                        String version = file.getName()
+                                .replace('b', 'B').replace('-', ' ');
+                        String update = version.split("\\.")[1];
+                        version = version + " (Swan Lake Update " + update + ")";
+                        String executableName = OSUtils.isWindows() ? WINDOWS_BAL_EXECUTABLE : UNIX_BAL_EXECUTABLE;
+                        Path sdkPath =
+                                Paths.get(file.getAbsolutePath(), BAL_EXECUTABLE_DIR, executableName).normalize();
+                        if (isValidSdk(sdkPath.toString(), version)) {
+                            sdkList.add(new BallerinaSdk(sdkPath.toString(), version));
+                        }
+                    }
+                }
+            }
+            return sdkList;
+        } catch (Exception e) {
+            LOG.error("Error occurred while getting the Ballerina SDKs", e);
+            return sdkList;
+        }
+    }
+
+    public static String findBalDistFolder(String initialPath) {
+        try {
+            File current = new File(Paths.get(initialPath).normalize().toString());
+
+            while (current != null) {
+                if (current.getName().toLowerCase().contains(BAL_DIST_DIR_NAME_START)) {
+                    return BallerinaSdkUtils.getNormalizedPath(current.getAbsolutePath());
+                }
+                current = current.getParentFile();
+            }
+            LOG.info("Ballerina distribution folder is not found");
+            return EMPTY;
+        } catch (Exception e) {
+            LOG.error("Error occurred while finding the Ballerina distribution folder", e);
+            return EMPTY;
+        }
+    }
+
+    public static String getBalBatFromDist(String distPath) {
+        try {
+            Path path = Paths.get(distPath).normalize();
+            String lastElement = path.getFileName().toString();
+
+            String executableName = OSUtils.isWindows() ? WINDOWS_BAL_EXECUTABLE : UNIX_BAL_EXECUTABLE;
+
+            if (BAL_EXECUTABLE_DIR.equals(lastElement)) {
+                return path.resolve(executableName).toString();
+            } else {
+                return path.resolve(Paths.get(BAL_EXECUTABLE_DIR, executableName)).toString();
+            }
+        } catch (Exception e) {
+            LOG.error("Error occurred while getting the Ballerina bat from the distribution", e);
+            return EMPTY;
+        }
+    }
+
+    public static String getNormalizedPath(String path) {
+        try {
+            if (path.isEmpty()) {
+                LOG.info("Path is empty");
+                return EMPTY;
+            }
+            return Paths.get(path).normalize().toString();
+        } catch (Exception e) {
+            LOG.error("Error occurred while normalizing the path", e);
+            return EMPTY;
```

Comment:
Better to rename the constant as `EMPTY_STRING`
```suggestion
            return EMPTY_STRING;
```